src/Provider/Nominatim/Nominatim.php

```diff
@@ -210,7 +210,16 @@ private function executeQuery(string $url, string $locale = null): string
             $url = sprintf('%s&accept-language=%s', $url, $locale);
         }
 
-        return $this->getUrlContents($url);
+        $request = $this->getRequest($url);
+
+        if (isset($_SERVER['HTTP_USER_AGENT'])) {
+            $request = $request->withHeader('User-Agent', $_SERVER['HTTP_USER_AGENT']);
+        }
+        if (isset($_SERVER['HTTP_REFERER'])) {
+            $request = $request->withHeader('Referer', $_SERVER['HTTP_REFERER']);
```

Comment:
Is Referer need?

Reply:
User-Agent **or** Referer needs to be set :
> Provide a valid HTTP Referer or User-Agent identifying the application (stock User-Agents as set by http libraries will not do).